### PR TITLE
Fix parsing of npm purls with namespace #93

### PR DESCRIPTION
**Pull request by TG1999**
Signed-off-by: Tushar Goel <tushar.goel.dav@gmail.com>

### CODE REVIEW

**Comment by pombredanne** [comment]
You should update the GH workflows to drop 3.6 and add 3.11

**Comment by TG1999** [comment]
Thanks @pombredanne the CI tests are now running fine.

File: src/packageurl/__init__.py [hunk 1]
```diff
@@ -374,6 +374,36 @@ def from_string(cls, purl):
             raise ValueError(msg)
 
         path = path.lstrip("/")
+
+        # NPM purl have a namespace in the path
+        # and the namespace in an npm purl is
+        # different from others because it starts with `@`
+        # so we need to handle this case separately
+        if type == "npm" and path.startswith("@"):
+            namespace = ""
```
Comment (pombredanne) [inline src/packageurl/__init__.py:383]: This is not needed

File: src/packageurl/__init__.py [hunk 2]
```diff
@@ -374,6 +374,36 @@ def from_string(cls, purl):
...
+            name = ""
```
Comment (pombredanne) [inline src/packageurl/__init__.py:384]: This is not needed either

File: src/packageurl/__init__.py [hunk 3]
```diff
@@ -374,6 +374,36 @@ def from_string(cls, purl):
...
+            namespace, sep, name_version = path.partition("/")
```
Comment (pombredanne) [inline src/packageurl/__init__.py:385]:
```suggestion
            namespace, sep, path = path.partition("/")
```
and then just do not special case anymore and the new code added below

**Review by pombredanne — CHANGES_REQUESTED** [review]
Thanks for your consideration.

**Comment by TG1999** [comment]
We need name to be declared here https://github.com/package-url/packageurl-python/pull/106/files#diff-1ac2b399571df74798b2fbfab357ffa8cce45a332f1e69b6f1ccf86811d44ba6R394 , so we can error out here https://github.com/package-url/packageurl-python/pull/106/files#diff-1ac2b399571df74798b2fbfab357ffa8cce45a332f1e69b6f1ccf86811d44ba6R402

**Review by pombredanne — APPROVED** [review]
LGTM!
Thanks

**Pull request merged**